server/src/controller/userController.js

```diff
@@ -64,7 +67,8 @@ const getMe = async (req, res) => {
       roles,
     });
   } catch (err) {
-    res.status(400).json({ error: err.message });
+    //send 401 Unauthorized status code for getMe fail(user not found)
+    res.status(401).json({ error: err.message });
```

Comment:
In here,
```
const { _id, name, email, postalCode, roles } = await User.findById(
      user.id
    );
```

or 

```
const user = await User.findById(req.user.id);
```

will be the cause for the catch block. Above two pieces of code are used to find the user. That's why I used 401. Please correct me If I'm wrong here. Thank you very much for the review.

Reply:
I didn't update this fix yet. Please let me know.

Reply:
I was talking about here on lines 55-56:

```
//send 401 Unauthorized status code for getMe fail(user not found)
    if (!user) return res.status(401).json({ msg: "User does not exist" });
```

Reply:
Yeah in there we can use 500. I used 401 because the user variable is null at that point. But as you mentioned we can use 500 as well. I will change and commit. Thank you.

Reply:
I was saying put the 500 in the catch. Why don't we wait and see what @lugenx  thinks.

Reply:
I thought the catch clause executed as I mentioned before. In there,
```
User.findById()
```
is the only cause for the catch clause.
 That's why I used 401 on both. Yeah, let's wait.
 Thanks again for your support.

Reply:
Of course! Apologies for any confusion

Reply:
No not at all. So is it ok to put 401 on both or should we wait for @lugenx?

Reply:
Are you in the discord? If not you can hop in there so we can discuss

Reply:
Yeah, I'm in discord.
DenZ#0031
Please send a message.

Reply:
Great! Join the [Ecohabit Server](https://discord.gg/2HxMurvQ) and we can chat there

Reply:
i might be missing something, but the first time (line 53), we check if the user doesn't exist and send an unauthorized error. if the code on 55-56 runs and an error occurs, that means the user already exists in the DB otherwise it would never run, so the issue is not authorization and something else. so 500 makes more sense to me.